src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -26,42 +30,47 @@ public class KNNCodecUtil {
     // Java rounds each array size up to multiples of 8 bytes
     public static final int JAVA_ROUNDING_NUMBER = 8;
 
+    @AllArgsConstructor
     public static final class Pair {
-        public Pair(int[] docs, float[][] vectors, SerializationMode serializationMode) {
-            this.docs = docs;
-            this.vectors = vectors;
-            this.serializationMode = serializationMode;
-        }
-
         public int[] docs;
-        public float[][] vectors;
+        @Getter
+        @Setter
+        private long vectorAddress;
+        @Getter
+        @Setter
+        private int dimension;
         public SerializationMode serializationMode;
+
     }
 
     public static KNNCodecUtil.Pair getFloats(BinaryDocValues values) throws IOException {
         ArrayList<float[]> vectorList = new ArrayList<>();
         ArrayList<Integer> docIdList = new ArrayList<>();
+        long vectorAddress = 0;
+        int dimension = 0;
         SerializationMode serializationMode = SerializationMode.COLLECTION_OF_FLOATS;
         for (int doc = values.nextDoc(); doc != DocIdSetIterator.NO_MORE_DOCS; doc = values.nextDoc()) {
             BytesRef bytesref = values.binaryValue();
             try (ByteArrayInputStream byteStream = new ByteArrayInputStream(bytesref.bytes, bytesref.offset, bytesref.length)) {
                 serializationMode = KNNVectorSerializerFactory.serializerModeFromStream(byteStream);
                 final KNNVectorSerializer vectorSerializer = KNNVectorSerializerFactory.getSerializerByStreamContent(byteStream);
                 final float[] vector = vectorSerializer.byteToFloatArray(byteStream);
+                dimension = vector.length;
```

Comment:
in this case `dimension` will be taken from the last value from the binary doc values. Just making sure it's intentional and all those are always identical for all elements

Reply:
Yes that is the understanding. We don't get vectors of different dimensions.